internal/services/dispatcher/server.go

```diff
@@ -553,14 +553,7 @@ func (s *DispatcherImpl) Heartbeat(ctx context.Context, req *contracts.Heartbeat
 		return nil, err
 	}
 
-	// if we haven't seen the dispatcher for 6 seconds (one interval plus latency), reject the heartbeat as the client
-	// should reconnect
-	if worker.DispatcherLastHeartbeatAt.Time.Before(time.Now().Add(-6 * time.Second)) {
```

Comment:
Removed -- I don't think this check is useful and it's causing problems when the DB is under high load.